# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/PresShell.h
# Commit: c428c3ad7725
# Full Hash: c428c3ad7725593f33b4ea2da9851be31be0c1df
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2022-11-28 15:41:50
# Regressor Bug: 1791759
# File Overlap Count: 1
# Description:
#   Bug 1791759 - Add support for `content-visibility: auto` r=emilio
#   
#   This change adds support for `content-visibilty: auto` as well as
#   showing and hiding content based on the relevancy of the content as
#   defined in the specification. Changes to relevancy are handled by
# ==============================================================================

diff -r ce553434720c -r c428c3ad7725 layout/base/PresShell.h
--- a/layout/base/PresShell.h	Mon Nov 28 09:45:39 2022 +0000
+++ b/layout/base/PresShell.h	Mon Nov 28 09:53:06 2022 +0000
@@ -13,6 +13,7 @@
 
 #include <stdio.h>  // for FILE definition
 #include "FrameMetrics.h"
+#include "LayoutConstants.h"
 #include "TouchManager.h"
 #include "Units.h"
 #include "Visibility.h"
@@ -1749,6 +1750,13 @@
    */
   bool IsForcingLayoutForHiddenContent(const nsIFrame*) const;
 
+  void RegisterContentVisibilityAutoFrame(nsIFrame* aFrame) {
+    mContentVisibilityAutoFrames.Insert(aFrame);
+  }
+
+  void UpdateRelevancyOfContentVisibilityAutoFrames();
+  void ScheduleContentRelevancyUpdate(ContentRelevancyReason aReason);
+
  private:
   ~PresShell();
 
@@ -3009,6 +3017,12 @@
 
   nsTHashSet<nsIContent*> mHiddenContentInForcedLayout;
 
+  nsTHashSet<nsIFrame*> mContentVisibilityAutoFrames;
+
+  // The type of content relevancy to update the next time content relevancy
+  // updates are triggered for `content-visibility: auto` frames.
+  ContentRelevancy mContentVisibilityRelevancyToUpdate;
+
   nsCallbackEventRequest* mFirstCallbackEventRequest = nullptr;
   nsCallbackEventRequest* mLastCallbackEventRequest = nullptr;
 